instrumentor/internal/webhook_env_injector/webhook_env_injector.go

```diff
@@ -38,6 +40,53 @@ func InjectOdigosAgentEnvVars(ctx context.Context, logger logr.Logger, podWorklo
 		return
 	}
 
+	injectionMethod := config.AgentEnvVarsInjectionMethod
+	if injectionMethod == nil {
+		// we are reading the effective config which should already have the env injection method resolved or defaulted
+		logger.Error(errors.New("env injection method is not set in ODIGOS config"), "Skipping Injection of ODIGOS agent")
+		return
+	}
+
+	// check if odigos loader should be used
+	if *injectionMethod == common.LoaderEnvInjectionMethod || *injectionMethod == common.LoaderFallbackToPodManifestInjectionMethod {
+		odigosLoaderPath := filepath.Join(k8sconsts.OdigosAgentsDirectory, commonconsts.OdigosLoaderName)
+
+		manifestValExits := getContainerEnvVarPointer(&container.Env, commonconsts.LdPreloadEnvVarName) != nil
+		runtimeDetailsVal, foundInInspection := getEnvVarFromRuntimeDetails(runtimeDetails, commonconsts.LdPreloadEnvVarName)
+		ldPreloadUnsetOrExpected := !foundInInspection || runtimeDetailsVal == odigosLoaderPath
```

Comment:
should we use `strings.Contains()` instead of equals to cover cases where our value is added to some other value?